### PR TITLE
Refactor Subscribers:subscribe to Doctrine [MAILPOET-3032]

### PR DESCRIPTION
**Pull request by lysyjan**
[MAILPOET-3032]

[MAILPOET-3032]: https://mailpoet.atlassian.net/browse/MAILPOET-3032

### CODE REVIEW

**Review by costasovo — COMMENTED** [review]
This is a huge amount of work! Great job! I have a couple of questions. See comments.

File: lib/Subscribers/SubscriberSaveController.php [hunk 1]
```diff
@@ -194,6 +194,9 @@ public function createOrUpdate(array $data, ?SubscriberEntity $subscriber): Subs
     $confirmedAt = isset($data['confirmed_at']) ? Carbon::createFromFormat('Y-m-d H:i:s', $data['confirmed_at']) : null;
     if ($confirmedAt) $subscriber->setConfirmedAt($confirmedAt);
 
+    // wipe any unconfirmed data at this point
+    $subscriber->setUnconfirmedData(null);
```
[the remark this answers is not in the document]
Comment (lysyjan) [reply lib/Subscribers/SubscriberSaveController.php:198]: This is behavior from the old model

File: lib/Subscribers/SubscriberActions.php [hunk 2]
```diff
@@ -23,77 +24,95 @@ class SubscriberActions {
...
-        return $subscriber;
-      }
-
-      $subscriber = Subscriber::findOne($subscriber->id);
+      $subscriber = $this->subscriberSaveController->createOrUpdate($subscriberData, $subscriber);
     } else {
       // store subscriber data to be updated after confirmation
-      $subscriber->setUnconfirmedData($subscriberData);
-      $subscriber->setExpr('updated_at', 'NOW()');
+      $unconfirmedData = $this->subscriberSaveController->filterOutReservedColumns($subscriberData);
```
[the remark this answers is not in the document]
Comment (lysyjan) [reply lib/Subscribers/SubscriberActions.php:80]: We modify `$subscriberData` between those lines.

File: lib/CustomFields/CustomFieldsRepository.php [hunk 3]
```diff
@@ -18,6 +18,11 @@ protected function getEntityClassName() {
    * @return CustomFieldEntity
    */
   public function createOrUpdate($data) {
+    // set name as label by default
+    if (empty($data['params']['label']) && isset($data['name'])) {
+      $data['params']['label'] = $data['name'];
+    }
```
[the remark this answers is not in the document]
Comment (lysyjan) [reply lib/CustomFields/CustomFieldsRepository.php:24]: This is old functionality copied from the old model. API tests were failing.

**Review by costasovo — APPROVED** [review]
Good job!

**Review by veljkho — APPROVED** [review]

**Pull request merged**